packages/components/src/components/domainproperties/list/ListPropertiesPanel.tsx

```diff
@@ -80,10 +82,17 @@ class ListPropertiesPanelImpl extends React.PureComponent<Props, State> {
         updateDomainPanelClassList(prevProps.useTheme, undefined, PROPERTIES_HEADER_ID);
     }
 
-    setIsValid(): void {
-        const { model } = this.props;
-        const isValid = model && model.hasValidProperties();
-        this.setState(() => ({isValid}));
+    setIsValid(newModel?: ListModel): void {
+        const { model, onChange } = this.props;
+        const updatedModel = newModel || model;
+        const isValid = updatedModel && updatedModel.hasValidProperties();
+        this.setState(() => ({isValid}),
+            () => {
+                // Issue 39918: only consider the model changed if there is a newModel param
+                if (newModel) {
+                    onChange(updatedModel)
```

Comment:
Not quite. The state for this component is whether or not the panel is valid. After that state is updated we are calling back to the parent component's onChange handler with the module changes (which is a prop to this component). I agree that the function name is misleading in that regard and I'l look at updating it (likely in the next round of designer changes since this "setIsValid" is a common function name for assay, list, and data class designers and I'd rather update them all at once.